File: mobile/src/main/java/net/fred/feedex/view/AutoSummaryListPreference.java
/**
 * Flym
 * <p/>
 * Copyright (c) 2012-2015 Frederic Julian
 * <p/>
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * <p/>
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * <p/>
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package net.fred.feedex.view;

import android.content.Context;
import android.util.AttributeSet;

public class AutoSummaryListPreference extends android.preference.ListPreference {

    public AutoSummaryListPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onDialogClosed(boolean positiveResult) {
        super.onDialogClosed(positiveResult);
        if (positiveResult) {
            setSummary(getEntry());
        }
    }

    @Override
    protected void onSetInitialValue(boolean restoreValue, Object defaultValue) {
        super.onSetInitialValue(restoreValue, defaultValue);
        setSummary(getEntry());
    }
}

File: app/src/main/java/net/frju/flym/ui/views/AutoSummaryListPreference.kt
package net.frju.flym.ui.views

import android.content.Context
import android.util.AttributeSet

class AutoSummaryListPreference(context: Context, attrs: AttributeSet) : android.preference.ListPreference(context, attrs) {

	override fun onDialogClosed(positiveResult: Boolean) {
		super.onDialogClosed(positiveResult)
		if (positiveResult) {
			summary = entry
		}
	}

	override fun onSetInitialValue(restoreValue: Boolean, defaultValue: Any?) {
		super.onSetInitialValue(restoreValue, defaultValue)
		summary = entry
	}
}